projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,661 @@
+     * ```typescript
+     * const newValue: DateRange = { start: new Date("2/2/2012"), end: new Date("3/3/2013")};
+     * this.dateRangePicker.value = newValue;
+     * ```
+     */
+    public get value(): DateRange {
+        return this._value;
+    }
+
+    public set value(value: DateRange) {
+        this._value = value;
+        this._onChangeCallback(value);
+        this.updateInputs();
+    }
+
+    /**
+     * Selects a range of dates. If no `endDate` is passed, range is 1 day (only `startDate`)
+     *
+     * @example
+     * ```typescript
+     * public selectFiveDayRange() {
+     *  const today = new Date();
+     *  const inFiveDays = new Date(new Date().setDate(today.getDate() + 5));
+     *  this.dateRange.selectRange(today, inFiveDays);
+     * }
+     * ```
+     */
+    public selectRange(startDate: Date, endDate?: Date): void {
+        endDate = endDate ?? startDate;
+        const dateRange = [startDate, endDate];
+        this.calendar.selectDate(dateRange);
+        this.handleSelection(dateRange);
+    }
+
+    /** @hidden @internal */
+    public writeValue(value: DateRange): void {
+        this.value = value;
+    }
+
+    /** @hidden @internal */
+    public registerOnChange(fn: any): void {
+        this._onChangeCallback = fn;
+    }
+
+    /** @hidden @internal */
+    public registerOnTouched(fn: any): void { }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        switch (this.mode) {
+            case InteractionMode.DropDown:
+                this.attachOnKeydown();
+                this.applyFocusOnClose();
+                this.subscribeToDateEditorEvents();
+                break;
+            case InteractionMode.Dialog:
+                this.applyFocusOnClose();
+                break;
+        }
+        this.subscribeToToggleEvents();
+        this.configPositionStrategy();
+        this.configOverlaySettings();
+
+        const subsToClicked = () => {
+            this.$toggleClickNotifier.next();
+
+            this.toggleComponents.forEach(toggle => {
+                toggle.clicked.pipe(takeUntil(this.$toggleClickNotifier)).subscribe(() => this.open());
+            });
+        };
+
+        this.toggleComponents.changes.pipe(takeUntil(this.$destroy)).subscribe(() => subsToClicked());
+        subsToClicked();
+    }
+
+    /** @hidden */
+    public ngOnDestroy(): void {
+        this.$destroy.next();
+        this.$destroy.complete();
+        this.$toggleClickNotifier.next();
+        this.$toggleClickNotifier.complete();
+    }
+
+    /** @hidden @internal */
+    public handleOpening(event: CancelableEventArgs): void {
+        this.onOpening.emit(event);
+    }
+
+    /** @hidden @internal */
+    public handleClosing(event: CancelableEventArgs): void {
+        this.onClosing.emit(event);
+    }
+
+    /** @hidden @internal */
+    public onKeyDown(event: KeyboardEvent): void {
+        // TODO: make sure IgxDateTimeEditorDirective doesn't spin if ALT is pressed
```

Comment:
also TODO: handle `DEL` key. Now when `DEL` is clicked cursor does not go to next char => you cannot delete entire input with only clicking `DEL`, like you can do by clicking `Backspace`